lib/stpeprintf.h

```diff
@@ -0,0 +1,122 @@
+/*
+ * SPDX-FileCopyrightText:  2022 - 2023, Alejandro Colomar <alx@kernel.org>
+ *
+ * SPDX-License-Identifier:  BSD-3-Clause
+ */
+
+
+#ifndef SHADOW_INCLUDE_LIB_STPEPRINTF_H_
+#define SHADOW_INCLUDE_LIB_STPEPRINTF_H_
+
+
+#include <config.h>
+
+#include <stdarg.h>
+#include <stddef.h>
+#include <stdio.h>
+
+#include "defines.h"
+
+
+#if !defined(HAVE_STPEPRINTF)
```

Comment:
> Why don't you add one conditional sentence to build the complete file?

Hmmm, yes, that sounds reasonable.

> Or even better, can you make the compilation of this file conditional in the Makefile? This way we avoid any conditions in the code.

Since this is a header file, we can't really do that (we would need to include conditionally, but I don't like that.  I prefer unconditional inclusion, and then reducing the scope of conditionals to this file.  (This is connected to your other question; see there).

Reply:
Fixed.